app/src/main/java/com/ncnf/user/User.java

```diff
@@ -1,203 +1,240 @@
+        this.lastName = lastName;
+    }
 
-        return this.db.setDocument(this.path, initial_data);
+    public void setFriendsIds(List<String> friendsIds) {
+        this.friendsIds = friendsIds;
     }
 
-    private CompletableFuture<DatabaseResponse> update(String key, Object value){
-        return this.db.updateField(path, key, value);
+    public void setOwnedEventsIds(List<String> ownedEventsIds) {
+        this.ownedEventsIds = ownedEventsIds;
     }
 
-    public CompletableFuture<DatabaseResponse> getField(String field){
-        return this.db.getField(path, field);
+    public void setSavedEventsIds(List<String> savedEventsIds) {
+        this.savedEventsIds = savedEventsIds;
     }
 
-    public CompletableFuture<DatabaseResponse>  updateLastName(String name){
-        return this.update(LAST_NAME_KEY, name);
+    public void setBirthDate(Date birthDate) {
+        this.birthDate = birthDate;
     }
 
-    public CompletableFuture<DatabaseResponse>  updateFirstName(String name){
-        return this.update(FIRST_NAME_KEY, name);
+    public void setNotifications(boolean notifications) {
+        this.notifications = notifications;
     }
 
-    public CompletableFuture<DatabaseResponse>  updateBirth(int year){
-        return this.update(BIRTH_YEAR_KEY, year);
+    public CompletableFuture<Boolean>  updateNotifications(boolean isEnabled) {
+        setNotifications(isEnabled);
+        return this.db.updateField(USERS_COLLECTION_KEY+uuid, NOTIFICATIONS_KEY, isEnabled);
     }
 
-    public CompletableFuture<DatabaseResponse>  updateNotifications(boolean isEnabled) {
-        return this.update(NOTIFICATIONS_KEY, isEnabled);
+    public CompletableFuture<Boolean>  updateNotificationsToken(String token) {
+        return this.db.updateField(USERS_COLLECTION_KEY+uuid, NOTIFICATIONS_TOKEN_KEY, token);
     }
 
-    public CompletableFuture<DatabaseResponse>  updateNotificationsToken(String token) {
-        return this.update(NOTIFICATIONS_TOKEN_KEY, token);
+    public CompletableFuture<Boolean> saveUserToDB(){
+        return this.db.setDocument(USERS_COLLECTION_KEY + uuid, this);
     }
 
-    public CompletableFuture<DatabaseResponse> loadUserFromDB(){
-        return this.db.getData(this.path);
+    public CompletableFuture<User> loadUserFromDB(){
+        CompletableFuture<User> futureUser = this.db.getDocument(USERS_COLLECTION_KEY + uuid, User.class);
+
+        return futureUser.thenApply(response -> {
+            User user = response;
+
+            this.username = user.getUsername();
+            this.email = user.getEmail();
+            this.firstName = user.getFirstName();
+            this.lastName = user.getLastName();
+            this.friendsIds = user.getFriendsIds();
+            this.ownedEventsIds = user.getOwnedEventsIds();
+            this.savedEventsIds = user.getSavedEventsIds();
+            this.birthDate = user.getBirthDate();
+            this.notifications = user.getNotifications();
```

Comment:
To reuse the UserBuilder, we could change the method `User toObject(uuid, Map)` to `void toObject(User, uuid, Map)` such that it could also be used here `toObject(this, ..., toMap(user))`. 
Don't know if it is worth it.

Reply:
Yeah maybe this can be done, but I think this will change shortly. I took some time with Dorian and she things have to change.